### ISSUE TITLE
Crashes while teleporting vehicles

### ISSUE THREAD

**Issue by ghost** [MattJeanes/Safe-Space#20]
If a simfphys vehicle passes through the door to the safe space the wheels trying to connect to the car will crash the server, I'm assuming this has to do with the absurd velocity and might also affect SCars.

I've lowered the server's angular velocity and overall max velocity to below the engine limits to avoid crashing but it hasn't fixed this issue.

Is there any available solution to this such as preventing certain entities from being teleported?

**Comment by LoganDark** [comment]
@Zafiro2232 I don't believe the Safe Space can teleport any contraption all at once, which is sad, because it also screws up SCars and anything else with more than one part.

**Comment by AmyJeanes** [comment]
Hey, as Logan said it just teleport one prop at a time it doesn't check for welded contraptions. Unfortunately I don't work with GMod much these days but if someone was to write up some improvements to it and send them to me in a pull request or whatever I'd be happy to test it out and release it if it's all good for everyone to enjoy.

**Comment by LoganDark** [comment]
I'm not much of a GMod addon dev, unfortunately, to be honest... I've only made around a couple addons, one of which being a modification to a very small addon (that stopped it from instantly crashing the server by screwing up prediction, jeez, that addon dev was very careless).

**Comment by linkjay** [comment]
This should be very possible. You can obtain constraints of any entity and then teleport them accordingly with the parent prop.

http://wiki.garrysmod.com/page/constraint/GetAllConstrainedEntities

**Comment by AmyJeanes** [comment]
Sure but when do you teleport it? Most of the contraption would end up behind the portal entrance, never thought of a good answer to it for multiple props even Portal doesn't do those correctly

**Comment by LoganDark** [comment]
You'd probably have to recreate the entire constraint system to work through portals, which, for obvious reasons, will introduce many compatibility issues...
